rust/foxglove/src/context.rs

```diff
@@ -218,9 +229,17 @@ impl Context {
         Arc::clone(LazyContext::get_default())
     }
 
-    /// Returns a channel builder for a channel in this context.
-    pub fn channel_builder(self: &Arc<Self>, topic: impl Into<String>) -> ChannelBuilder {
-        ChannelBuilder::new(topic).context(self)
+    /// Creates a builder for a channel in this context. Results in a
+    /// [`FoxgloveError::DuplicateChannel`] if a channel with the same topic already exists.
```

Comment:
This should be allowed now? Seems inconsistent.

Maybe it's to support log!, but maybe we could think of a better API for that anyway.

Reply:
I kept it because it seemed like it was steering people in the right direction, but I agree with you; I think I should update this to be consistent and rely on warnings and documentation.